### PR TITLE
use azurite for integration tests

### PR DESCRIPTION
**Pull request by ctaggart**
This switches the integration tests from the AzureStorageEmulator that is erroring out on us in #284 to Azurite.

> The Azure Storage Emulator is no longer being actively developed. Azurite is the Storage Emulator platform going forward. Azurite supersedes the Azure Storage Emulator.
https://docs.microsoft.com/en-us/azure/storage/common/storage-use-emulator

FYI, all of these tests pass without `azurite_workaround` feature enabled.

### CODE REVIEW

**Comment by arnodb** [comment]
Hi @ctaggart, `azurite_workaround` becomes necessary when you try to get a blob properties, it does not return any `x-ms-creation-time` header. (That is with Azurite 3.11.0)

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -85,7 +85,7 @@ jobs:
 
   test-integration:
     name: Integration Tests
-    runs-on: windows-2019
```
Comment (rylev) [inline .github/workflows/build.yml:88]: Should we just move to testing both platforms?

File: sdk/storage/src/table/clients/partition_key_client.rs [hunk 2]
```diff
@@ -67,6 +67,8 @@ impl PartitionKeyClient {
 
 #[cfg(test)]
 #[cfg(feature = "test_integration")]
+// TODO #297 enable test once transactions are working in Azurite
+#[cfg(feature = "azurite_todo")]
```
Comment (rylev) [inline sdk/storage/src/table/clients/partition_key_client.rs:71]: Are we doing this todo as a feature so we can easily re-enable for testing? I guess I would assume we'd just use `#[ignore = "enable when azure is working with transactions"]`

**Review by rylev — APPROVED** [review]
Looks good!

File: .github/workflows/build.yml (hunk 1 above)
Comment (ctaggart) [reply .github/workflows/build.yml:88]: Sure, but another PR preferred.

File: sdk/storage/src/table/clients/partition_key_client.rs (hunk 2 above)
Comment (ctaggart) [reply sdk/storage/src/table/clients/partition_key_client.rs]: Using `ignore` makes more sense to me. I updated the PR. FYI, I did have to add `unused_attributes` to `RUSTFLAGS: --deny warnings --allow unused_attributes`.

**Pull request merged**